src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -769,6 +774,19 @@ private function registerRequestConfiguration(array $config, ContainerBuilder $c
         }
     }
 
+    private function registerEncryptedSecretsConfiguration(array $config, ContainerBuilder $container)
+    {
+        $secretsFile = $config['encrypted_secrets']['secrets_file'];
+        $publicKeyFile = $config['encrypted_secrets']['public_key_file'];
+        $privateKeyFile = $config['encrypted_secrets']['private_key_file'];
+        $secretsHandler = new JweHandler($secretsFile, $publicKeyFile, $privateKeyFile);
```

Comment:
This part will "smell" on code review ;). We never want to create objects manually. Instead, register this as a service in some new `secrets.xml` file, but leave the 3 args empty. Then, in here, read the Definition for that service and set the 3 arguments specifically. This is another reason to have only the "config" way of making secrets - it allows us to register this one service properly, and avoid `new JweHandler` cllas

Reply:
ahh okay i didn't realize you could leave them blank, thanks! that cleans up a bunch